### ISSUE TITLE
Aria Role = list in slot

### ISSUE THREAD

**Issue by alethut** [vaadin/vcf-nav#6]
according to W3 no role or aria-* attributes is allowed in the tag slot.
I think the 'role' = 'list' should go in the vcf-nav tag bacause

**Comment by jouni** [comment]
Yes, thanks for the reminder. I wasn't aware of this spec when I did the implementation, and I just wanted to avoid as many unnecessary elements as possible (small performance benefit I believe). And in my testing, it seemed to work just fine with the role on the slot element. Granted, I only tested this on macOS and VoiceOver, so it might be an issue on some other platforms.

Since the spec is clear on this, we should not rely on this behavior (that there are no unwanted side-effects from using `role` on `<slot>` elements), and the official component will introduce an extra element in the shadow DOM that has the list role.

**Comment by rolfsmeds** [comment]
Note to fixer: there is a `<slot>` with role and part name also in the nav-_item_ element, so those should also be moved to a separate wrapper element around the `<slot>`
